File: library/src/main/java/me/proxer/library/api/chat/ChatApi.java
package me.proxer.library.api.chat;

import retrofit2.Retrofit;

/**
 * Api for the Chat class.
 *
 * @author Ruben Gees
 */
public final class ChatApi {

    private final InternalApi internalApi;

    /**
     * Only for internal use.
     */
    public ChatApi(final Retrofit retrofit) {
        this.internalApi = retrofit.create(InternalApi.class);
    }

    /**
     * Returns the respective endpoint.
     */
    public PublicChatRoomsEndpoint publicRooms() {
        return new PublicChatRoomsEndpoint(internalApi);
    }

    /**
     * Returns the respective endpoint.
     */
    public ChatMessagesEndpoint messages(final String roomId) {
        return new ChatMessagesEndpoint(internalApi, roomId);
    }
}

File: library/src/main/kotlin/me/proxer/library/api/chat/ChatApi.kt
package me.proxer.library.api.chat

import retrofit2.Retrofit

/**
 * Api for the Chat class.
 *
 * @author Ruben Gees
 */
class ChatApi internal constructor(retrofit: Retrofit) {

    private val internalApi = retrofit.create(InternalApi::class.java)

    /**
     * Returns the respective endpoint.
     */
    fun publicRooms(): PublicChatRoomsEndpoint {
        return PublicChatRoomsEndpoint(internalApi)
    }

    /**
     * Returns the respective endpoint.
     */
    fun userRooms(): UserChatRoomsEndpoint {
        return UserChatRoomsEndpoint(internalApi)
    }

    /**
     * Returns the respective endpoint.
     */
    fun messages(roomId: String): ChatMessagesEndpoint {
        return ChatMessagesEndpoint(internalApi, roomId)
    }

    /**
     * Returns the respective endpoint.
     */
    fun sendMessage(roomId: String, message: String): SendChatMessageEndpoint {
        return SendChatMessageEndpoint(internalApi, roomId, message)
    }

    /**
     * Returns the respective endpoint.
     */
    fun reportMessage(messageId: String, message: String): ReportChatMessageEndpoint {
        return ReportChatMessageEndpoint(internalApi, messageId, message)
    }

    /**
     * Returns the respective endpoint.
     */
    fun roomUsers(roomId: String): ChatRoomUsersEndpoint {
        return ChatRoomUsersEndpoint(internalApi, roomId)
    }
}
